airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_excel_parser.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import datetime
+from io import BytesIO
+from unittest.mock import Mock, mock_open, patch
+
+import pandas as pd
+import pytest
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import ExcelFormat, FileBasedStreamConfig, ValidationPolicy
+from airbyte_cdk.sources.file_based.exceptions import ConfigValidationError, RecordParseError
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.file_types.excel_parser import ExcelParser
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+
+
+@pytest.fixture
+def mock_stream_reader():
+    return Mock(spec=AbstractFileBasedStreamReader)
+
+
+@pytest.fixture
+def mock_logger():
+    return Mock()
+
+
+@pytest.fixture
+def file_config():
+    return FileBasedStreamConfig(
+        name="test.xlsx",
+        file_type="excel",
+        format=ExcelFormat(sheet_name="Sheet1"),
+        validation_policy=ValidationPolicy.emit_record
+    )
+
+
+@pytest.fixture
+def remote_file():
+    return RemoteFile(uri="s3://mybucket/test.xlsx", last_modified=datetime.datetime.now())
+
+
+@pytest.fixture
+def sample_excel_data():
+    data = {
+        'A': [1, 2, 3],
+        'B': ['2023-01-01', '2023-02-01', '2023-03-01'],
+        'C': [True, False, True]
+    }
+    df = pd.DataFrame(data)
+    with BytesIO() as buffer:
+        with pd.ExcelWriter(buffer, engine='xlsxwriter') as writer:
```

Comment:
Why do we use a different engine than [here](https://github.com/airbytehq/airbyte/pull/43346/files#diff-aa4686a61ca7e6f6525983b4aa54e329a0eea38502225700d8b11398e6cbc8b0R56)?

Reply:
in the code we use a reading engine, but this is a writing engine